open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,736 @@
+                    return amount;
+                }
+
+                JSONObject currency = validateCurrencyPeriodicLimit(controlParameters);
+                if (!(Boolean.parseBoolean(currency.
+                        getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                    return currency;
+                }
+
+                JSONObject validationResults = validatePeriodAlignment(limit);
+                if (!(Boolean.parseBoolean(validationResults.
+                        getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                    return validationResults;
+                }
+
+                JSONObject periodType = validatePeriodType(limit);
+                if (!(Boolean.parseBoolean(periodType.
+                        getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                    return periodType;
+                }
+            }
+        } else {
+            // If periodic limits key is missing, return an error
+            log.error(ErrorConstants.MISSING_PERIOD_LIMITS);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.MISSING_PERIOD_LIMITS);
+        }
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Validates the Currency in periodic limits in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response.
+     */
+    public static JSONObject validateCurrencyPeriodicLimit(JSONObject controlParameters) {
+
+        JSONObject validationResponse = new JSONObject();
+
+        JSONArray periodicLimits = (JSONArray) controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+        JSONObject currency = validateAmountCurrencyPeriodicLimits((JSONArray) periodicLimits,
+                ConsentExtensionConstants.CURRENCY);
+        if (!(Boolean.parseBoolean(currency.
+                getAsString(ConsentExtensionConstants.IS_VALID)))) {
+            return currency;
+        }
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+
+    /**
+     * Validates the Amount in periodic limits in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response.
+     */
+    public static JSONObject validateAmountPeriodicLimit(JSONObject controlParameters) {
+
+        JSONObject validationResponse = new JSONObject();
+
+        JSONArray periodicLimits = (JSONArray) controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+
+        JSONObject amount = validateAmountCurrencyPeriodicLimits((JSONArray) periodicLimits,
```

Comment:
Here `amount` means it's validation response right? Therefor shall we use naming to indicate that? Ex : `amountValidationResponse` WDUT?

Reply:
Fixed in [74be5dc]